File: Python/Solution_392.py
class Solution_392(object):
    def isSubsequence(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s) == 0:
            return True
        
        sPointer, tPointer = 0, 0
        
        while tPointer < len(t) :
            if s[sPointer] == t[tPointer]:
                sPointer += 1
                if sPointer == len(s):
                    return True
            tPointer += 1
            
        return False

# Driver code to test the function
if __name__ == "__main__":
    solution = Solution_392()

    # Example 1
    s1 = "abc"
    t1 = "ahbgdc"
    
    print(solution.isSubsequence(s1, t1))  # Output: True
    
    # Example 2
    s2 = "axc"
    t2 = "ahbgdc"
    
    print(solution.isSubsequence(s1, t1))  # Output: False

File: Java/Solution_392.java
public class Solution_392 {
    public boolean isSubsequence(String s, String t) {
        if (s.length() == 0) return true;

        int sPointer = 0, tPointer = 0;

        while (tPointer < t.length()) {
            if (s.charAt(sPointer) == t.charAt(tPointer)) {
                sPointer++;
                if (sPointer == s.length()) {
                    return true;
                }
            }
            tPointer++;
        }

        return false;
    }

    public static void main(String[] args){
        Solution_392 sol = new Solution_392();

        String s1 = "abc";
        String t1 = "ahbgdc";
        
        System.out.println(sol.isSubsequence(s1, t1)); // Output: True

        String s2 = "axc";
        String t2 = "ahbgdc";
        
        System.out.println(sol.isSubsequence(s2, t2));   // Output: False
    }
}
